openghg/transform/emissions/_edgar.py

```diff
@@ -0,0 +1,1066 @@
+                    raise ValueError("Latitude values should not be specified"
+                                    f" when using pre-defined domain {domain}"
+                                     " (values don't match).")
+            else:
+                lat_out = lat_domain
+
+            if lon_out is not None:
+                if not np.array_equal(lon_domain, lon_out):
+                    raise ValueError("Longitude values should not be specified"
+                                    f" when using pre-defined domain {domain}"
+                                     " (values don't match).")
+            else:
+                lon_out = lon_domain
+
+            if lon_out.max() > 180 or lon_out.min() < -180:
+                raise ValueError("Invalid domain definition."
+                                 " Expected longitude in range: -180 - 180."
+                                f"Current longitude: {lon_out.min()} - {lon_out.max()}")
+
+    if lon_out is not None and (lon_out.max() > 180 or lon_out.min() < -180):
+        print("Converting longitude to stay within -180 - 180 bounds")
+        lon_converted = convert_longitude(lon_out)
+        lon_out = cast(Optional[ndarray], lon_converted)
+
+    return lat_out, lon_out
+
+
+def _edgar_known_versions() -> list:
+    """Define list of known versions for the EDGAR database"""
+    known_version = ["v432", "v50", "v6.0"]
+    return known_version
+
+
+def _check_readme_version(datapath: Optional[Path] = None,
+                          zippath: Optional[ZipFile] = None) -> Optional[str]:
+    """
+    Attempts to extract the edgar version from the associated "_readme.html"
+    file, if present.
+
+    Args:
+        datapath : Path to the folder containing the downloaded EDGAR files
+        zippath: Path to zipped archive file (direct from EDGAR)
+
+    Returns:
+        str : edgar version if found (None otherwise)
+    """
+
+    # Work out version if possible from readme
+    # All database versions so far may contain "_readme.html" file
+    # "v6.0"
+    #  - "TOTALS_nc.zip" is what is downloaded from website
+    #  - "_readme.html" title line: "<title>EDGAR v6.0_GHG (2021)</title>"
+    # "v5.0"
+    #  - "v50_CH4_1970_2015.zip" can be downloaded
+    #  - "_readme.html" title line: "<title>EDGAR v5.0 (2019)</title>"
+    # "v4.3.2"
+    #  - "v432_CH4_1970_2012.zip" can be downloaded
+    #  - "_readme.html" title line: "<title>EDGAR v4.3.2 (2017)</title>"
+
+    # Check for readme html file and, if present, extract version
+    readme_filename = "_readme.html"
+    if zippath is not None:
+        try:
+            # Cast extracted bytes to a str object
+            readme_data: Optional[str] = str(zippath.read(readme_filename))
+        except ValueError:
+            readme_data = None
+    elif datapath is not None:
+        try:
+            readme_filepath = os.path.join(datapath, readme_filename)
+            readme_data = open(readme_filepath, "r").read()
```

Comment:
I think it'd be better to do

```
readme_filepath = datapath.joinpath(readme_filename)
if readme_filepath.exists():
    readme_data = readme_filepath.read_text()
else:
    readme_data = None
```

At the moment we end up with an open file (I think at least) at the end of the function that isn't closed.